orchestration/src/main/java/com/sap/ai/sdk/orchestration/spring/OrchestrationSpringChatResponse.java

```diff
@@ -43,21 +42,18 @@ public class OrchestrationSpringChatResponse extends ChatResponse {
 
   @Nonnull
   static List<Generation> toGenerations(@Nonnull final LLMModuleResultSynchronous result) {
-    return result.getChoices().stream()
-        .map(OrchestrationSpringChatResponse::toAssistantMessage)
-        .map(Generation::new)
-        .toList();
+    return result.getChoices().stream().map(OrchestrationSpringChatResponse::toGeneration).toList();
   }
 
   @Nonnull
-  static AssistantMessage toAssistantMessage(@Nonnull final LLMChoice choice) {
-    final Map<String, Object> metadata = new HashMap<>();
-    metadata.put("finish_reason", choice.getFinishReason());
-    metadata.put("index", choice.getIndex());
+  static Generation toGeneration(@Nonnull final LLMChoice choice) {
+    val metadata = ChatGenerationMetadata.builder().finishReason(choice.getFinishReason());
+    metadata.metadata("index", choice.getIndex());
```

Comment:
index might belong somewhere else, to investigate...
The rest is drive by fixes